registry/repository.go

```diff
@@ -1,198 +1,117 @@
 package registry
 
 import (
-	"bytes"
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 
-	"github.com/notaryproject/notation-go"
-	"github.com/opencontainers/go-digest"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
-	artifactspec "github.com/oras-project/artifacts-spec/specs-go/v1"
 	"oras.land/oras-go/v2"
 	"oras.land/oras-go/v2/content"
-	"oras.land/oras-go/v2/registry"
-	"oras.land/oras-go/v2/registry/remote"
+	orasRegistry "oras.land/oras-go/v2/registry"
 )
 
 const (
 	maxBlobSizeLimit     = 32 * 1024 * 1024 // 32 MiB
 	maxManifestSizeLimit = 4 * 1024 * 1024  // 4 MiB
 )
 
-type RepositoryClient struct {
-	remote.Repository
+// repositoryClient implements Repository
+type repositoryClient struct {
+	orasRegistry.Repository
 }
 
-type SignatureManifest struct {
-	Blob        notation.Descriptor
-	Annotations map[string]string
-}
-
-// NewRepositoryClient creates a new registry client.
-func NewRepositoryClient(client remote.Client, ref registry.Reference, plainHTTP bool) *RepositoryClient {
-	return &RepositoryClient{
-		Repository: remote.Repository{
-			Client:    client,
-			Reference: ref,
-			PlainHTTP: plainHTTP,
-		},
+// NewRepository returns a new Repository
+func NewRepository(repo orasRegistry.Repository) Repository {
+	return &repositoryClient{
+		Repository: repo,
 	}
 }
 
 // Resolve resolves a reference(tag or digest) to a manifest descriptor
-func (c *RepositoryClient) Resolve(ctx context.Context, reference string) (notation.Descriptor, error) {
-	desc, err := c.Repository.Resolve(ctx, reference)
-	if err != nil {
-		return notation.Descriptor{}, err
-	}
-	return notationDescriptorFromOCI(desc), nil
+func (c *repositoryClient) Resolve(ctx context.Context, reference string) (ocispec.Descriptor, error) {
+	return c.Repository.Resolve(ctx, reference)
```

Comment:
If it is manifest descriptor only, should make it `c.Respoitory.Manifest().Resolve()`?